### PR TITLE
Re-enabled unit tests that were previously commented out

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes https://github.com/dotnet/templating/issues/2334
fixes https://github.com/dotnet/templating/issues/1726

### CODE REVIEW

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs [hunk 1]
```diff
@@ -1,482 +1,494 @@
...
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Core.Util;
+using Microsoft.TemplateEngine.TestHelper;
+using Microsoft.TemplateEngine.Utils;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Core.UnitTests
+{
+    public class MockFileStream : MemoryStream
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:18]: please consider implementing a destructor and implementing `IDisposable` to protect against leaking memory stream

https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern

Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:18]: consider sealing if you are not designing for inheritance

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs [hunk 2]
```diff
@@ -1,482 +1,494 @@
...
+    {
+        private readonly Action<byte[]> _onFlush;
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:20]: consider Reactive's Subject in such future cases

- https://rehansaeed.com/reactive-extensions-part1-replacing-events/
- https://docs.microsoft.com/en-us/previous-versions/dotnet/reactive-extensions/hh229173(v=vs.103)

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs [hunk 3]
```diff
@@ -1,482 +1,494 @@
...
+            {
+                _files.Remove(key);
+            }
+        }
+
+        public void FileCopy(string sourcePath, string targetPath, bool overwrite)
+        {
+            if (!overwrite && FileExists(targetPath))
+            {
+                throw new Exception($"File {targetPath} already exists");
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:155]: consider a closely related exception, such as [ioexception](https://docs.microsoft.com/en-us/dotnet/api/system.io.ioexception?view=netcore-3.1)

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs [hunk 4]
```diff
@@ -1,482 +1,494 @@
...
+            }
+
+            if (!FileExists(sourcePath))
+            {
+                throw new Exception($"File {sourcePath} doesn't exist");
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:160]: consider [filenotfoundexception](https://docs.microsoft.com/en-us/dotnet/api/system.io.filenotfoundexception?view=netcore-3.1)

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs [hunk 5]
```diff
@@ -1,482 +1,494 @@
...
+            throw new NotImplementedException();
+        }
+
+        public void SetFileAttributes(string file, FileAttributes attributes)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class MockGlobalRunSpec : IGlobalRunSpec
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:191]: please consider 1 type per file, under common circumstances

File: test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs (hunk 1 above)
Comment (vlada-shubina) [reply test/Microsoft.TemplateEngine.Core.UnitTests/OrchestratorTests.cs:18]: @donJoseLuis this code was just uncommented due to unit test was failing. This PR only addresses changes required to make test pass.
I can consider to refactor these tests, but imo it's quite low priority to other tasks at the moment.

**Review by AR-May — APPROVED** [review]

**Review by donJoseLuis — APPROVED** [review]

**Pull request merged**